tests/testthat/test-dparse.R

```diff
@@ -1,12 +1,29 @@
-test_that("ShortName, ClassName and Alias are unique across all distributions", {
-  d6 <- unlist(listDistributions()[,c("ShortName", "ClassName", "Alias")], use.names = F)
-  expect_equal(length(d6), unique(length(d6)))
+test_that("ShortName, ClassName and Alias are unique between distributions", {
+  d6 <- listDistributions()[,(ids = paste(tolower(ShortName), tolower(ClassName), tolower(Alias), sep = ", "))]
+  d6 <- strsplit(d6, ",")
+  # Unique across one class
+  d6 <- unlist(lapply(d6, function(x) unique(trimws(x))))
+  expect_equal(length(d6), length(unique(d6)))
 })
 
-test_that("Every distribution can be created", {
-  d6 <- unlist(listDistributions()[,c("ShortName", "ClassName", "Alias")], use.names = F)
-  d6 <- paste0(d6, "()")
-  expect_invisible(invisible(sapply(d6, dparse)))
+test_that("Every distribution is created with it's propper S3 class", {
+  # Get calls
+  d6 <- listDistributions()[,(ids = paste(tolower(ShortName), tolower(ClassName), tolower(Alias), sep = ", "))]
+  calls <- strsplit(d6, ",")
+  calls <- unlist(lapply(calls, function(x) trimws(x)))
+  calls <- calls[calls != ""]
+  calls <- paste0(calls, "()")
+  # Get classes
+  counts <- strsplit(d6, ",")
+  counts <- unlist(lapply(counts, length))
+  # Fix for Exponential distribution where Alias = "";
+  counts[16] <- counts[16] - 1
+  counts <- mapply(rep, unlist(listDistributions()[, "ClassName"], use.names = F), each = counts)
+  classes <- unlist(counts, use.names = F)
+  # Evaluate
+  for (i in seq_along(classes)) {
+    expect_s3_class(dparse(calls[i]), classes[i])
+  }
 })
```

Comment:
S3 not R6?? What am I missing here ? I think I'm, confused with that test is doing

Reply:
I used `testthat::expect_s3_class` to check that the returned distribution (e.g., the class) was correct. I now noticed that there is also a `distr6::expect_R6_class` function that works and is more suitable for these checks. It is now replaced.